cpp/include/raft/core/host_mdspan.hpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/mdspan.hpp>
+
+#include <raft/core/detail/accessor_mixin.hpp>
+
+namespace raft {
+
+template <typename AccessorPolicy>
+using host_accessor = detail::accessor_mixin<AccessorPolicy, true, false>;
+
+/**
+ * @brief std::experimental::mdspan with host tag to avoid accessing incorrect memory location.
+ */
+template <typename ElementType,
+          typename Extents,
+          typename LayoutPolicy   = layout_c_contiguous,
+          typename AccessorPolicy = std::experimental::default_accessor<ElementType>>
+using host_mdspan = mdspan<ElementType, Extents, LayoutPolicy, host_accessor<AccessorPolicy>>;
+
+namespace detail {
+
+template <typename T, bool B>
+struct is_host_mdspan : std::false_type {
+};
+template <typename T>
+struct is_host_mdspan<T, true> : std::bool_constant<T::accessor_type::is_host_accessible> {
```

Comment:
`is_host_mdspan<managed_mdspan<...>>::value` is `true` as well.  Is that OK?  If so, would you consider perhaps calling this trait something else, like `is_host_accessible_mdspan`?  It could be confusing that `is_host_mdspan<T>::value` could be `true` for things that aren't `host_mdspan`.